afs-cassandra/src/main/java/com/powsybl/afs/cassandra/DisplayTreeIssueBuilder.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.cassandra;
+
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.powsybl.afs.storage.check.FileSystemCheckIssue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+class DisplayTreeIssueBuilder {
+
+    private final TreeNode root = new TreeNode("display tree");
+    private final Map<String, TreeNode> nodes = new HashMap<>();
+    private final Map<String, NodeInfo> infos = new HashMap<>();
+
+    DisplayTreeIssueBuilder(ResultSet resultSet) {
+        for (Row row : resultSet) {
+            RowBean rowBean = new RowBean(row.getUUID(0).toString(),
+                    row.getString(1),
+                    row.getString(2),
+                    row.getUUID(3) == null ? null : row.getUUID(3).toString());
+            cacheNodeInfo(rowBean);
+            nodes.computeIfAbsent(rowBean.id, TreeNode::new);
+            if (rowBean.cId != null) {
+                nodes.computeIfAbsent(rowBean.cId, TreeNode::new);
+                nodes.get(rowBean.id).children.add(rowBean.cId);
+            }
+            if (row.getUUID(4) == null) {
+                root.children.add(rowBean.id);
```

Comment:
Looks like the tree is displayed multiple times because we don't check here if the node has already been added: 
if the root node of the filesystem has multiple children, it will appear multiple times in the table (once for itself, once for each children), so we will go through that code multiple times

Reply:
> Looks like the tree is displayed multiple times because we don't check here if the node has already been added:
> if the root node of the filesystem has multiple children, it will appear multiple times in the table (once for itself, once for each children), so we will go through that code multiple times

Fixed by 'Set'